src/main/java/de/tum/in/www1/artemis/config/ApiVersionFilter.java

```diff
@@ -0,0 +1,52 @@
+package de.tum.in.www1.artemis.config;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApiVersionFilter implements Filter {
```

Comment:
Unfortunately, most Server Tests fail with the current implementation. When removing the conformance to the `Filter` interface, all tests pass al the sudden.
```
public class ApiVersionFilter {
```

I cannot understand why this is the case. Maybe someone can help here?